Applications/Spire/Source/Ui/NavigationView.cpp

```diff
@@ -0,0 +1,301 @@
+    std::bind_front(&NavigationView::on_associative_value_current, this, index));
+  if(index == m_current_model->get_current()) {
+    on_current(index);
+  }
+}
+
+int NavigationView::get_tab_count() const {
+  return m_navigation_list->get_list_model()->get_size();
+}
+
+QString NavigationView::get_tab_label(int index) const {
+  if(index < 0 || index >= get_tab_count()) {
+    throw std::out_of_range("The index is out of range.");
+  }
+  return m_navigation_list->get_list_model()->get<QString>(index);
+}
+
+QWidget& NavigationView::get_page(int index) const {
+  if(index < 0 || index >= get_tab_count()) {
+    throw std::out_of_range("The index is out of range.");
+  }
+  return *m_stacked_widget->widget(index);
+}
+
+bool NavigationView::is_tab_enabled(int index) const {
+  if(index < 0 || index >= get_tab_count()) {
+    throw std::out_of_range("The index is out of range.");
+  }
+  return m_navigation_list->get_list_item(index)->isEnabled();
+}
+
+void NavigationView::set_tab_enabled(int index, bool is_enabled) {
+  if(index < 0 || index >= get_tab_count()) {
+    throw std::out_of_range("The index is out of range.");
+  }
+  if(!isEnabled()) {
+    return;
+  }
+  m_navigation_list->get_list_item(index)->setEnabled(is_enabled);
+  m_stacked_widget->widget(index)->setEnabled(is_enabled);
+  if(is_enabled) {
+    if(!m_navigation_list->isEnabled()) {
+      m_navigation_list->setEnabled(true);
+    }
+    if(!m_stacked_widget->currentWidget()->isEnabled()) {
+      m_current_model->set_current(index);
+    }
+  } else if(m_current_model->get_current() == index) {
+    auto new_index = [=] {
+      for(auto i = index + 1; i < get_tab_count(); ++i) {
+        if(m_navigation_list->get_list_item(i)->isEnabled()) {
+          return i;
+        }
+      }
+      for(auto i = index - 1; i > -1; --i) {
+        if(m_navigation_list->get_list_item(i)->isEnabled()) {
+          return i;
+        }
+      }
+      return -1;
+    }();
+    if(new_index > -1) {
+      m_current_model->set_current(new_index);
+    } else {
+      m_navigation_list->setEnabled(false);
+    }
+  }
+}
+
+void NavigationView::on_current(int index) {
+  if(index < 0 || index >= get_tab_count()) {
+    return;
+  }
+  m_associative_model.get_association(m_navigation_list->get_list_model()->
+    get<QString>(index))->set_current(true);
+}
+
+void NavigationView::on_list_submit(const std::any& submission) {
+  m_associative_model.get_association(
+    std::any_cast<QString>(submission))->set_current(true);
+}
+
+void NavigationView::on_list_current(const boost::optional<int>& current) {
+  if(current) {
+    m_stacked_widget->setFocusPolicy(Qt::TabFocus);
+    setTabOrder(m_stacked_widget, m_navigation_list->get_list_item(*current));
+    m_stacked_widget->setFocusPolicy(Qt::NoFocus);
+  }
+}
+
+void NavigationView::on_associative_value_current(int index, bool value) {
+  if(value) {
+    match(*m_navigation_list->get_list_item(index), Checked());
```

Comment:
A list item can't be checked, you can use ```Current``` if you'd like.

Reply:
The spec in the /drafts/navigation_view/navigation_tab uses Checked selector for styling the current NavigationTab of the NavigationView. I don’t think the Checked selector is the same as the Current selector in the NavigationView. When navigating to NavTab2 by pressing right arrow key from NavTab1, NavTab2 becomes the current item of the ListView to which the Current selector is applied, but the NavTab1 is still the current tab of the NavigationView which should have Checked selector to style SelectLine and Label.

Reply:
When you are asked to review a spec, you have a responsibility to ensure that the spec is correct. I am unaware that there is a spec in the drafts folder that has this that you were working off of and the spec makes no sense. The spec says that if the Label is checked then it is colored purpose, for example, but a Label can not be checked, there is no ability for a label to be checked under any circumstance (review its user flow).

The components.xd also says Box:SelectLine is purple when checked, but a Box can never be checked (review the model for Box).

If you are working off of a faulty spec, then you assume the responsibility of potentially submitting an implementation that is completely invalid. You are required to ensure that any specification you work off of is correct and take good care to ensure that specs are held to a high standard.

As it is the spec is not correct and I can not approve this change.